org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/internal/themes/ThemeBehaviourManager.java

```diff
@@ -0,0 +1,62 @@
+package org.eclipse.tm4e.ui.internal.themes;
+
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.tm4e.ui.TMUIPlugin;
+import org.eclipse.tm4e.ui.themes.IThemeBehaviour;
+
+public class ThemeBehaviourManager {
+
+	private static final String EXTENSION_POINT_NAME = TMUIPlugin.PLUGIN_ID + ".themeBehaviour";
+	private static ThemeBehaviourManager INSTANCE = new ThemeBehaviourManager();
+
+	private IThemeBehaviour behaviour;
+
+	private ThemeBehaviourManager() {
+	}
+
+	public static ThemeBehaviourManager getInstance() {
+		return INSTANCE;
+	}
+
+	public IThemeBehaviour getThemeBehaviour() {
+		return behaviour;
+	}
+
+	/**
+	 * Initialize manager
+	 */
+	public void init() {
+		IConfigurationElement[] configurationElements = Platform.getExtensionRegistry()
+				.getConfigurationElementsFor(EXTENSION_POINT_NAME);
+
+		if (configurationElements.length > 0) {
+			behaviour = getBehaviour(configurationElements[0]);
+		}
+
+		if (behaviour == null) {
+			behaviour = new DefaultThemeBehaviour();
+		}
+	}
+
+	/**
+	 * Get behaviour from extension point.
+	 * 
+	 * @param element
+	 *            configurationElement for extension point
+	 * @return behaviour from configuration or null if exception occured
+	 */
+	private IThemeBehaviour getBehaviour(IConfigurationElement element) {
+		IThemeBehaviour extendedBehaviour = null;
+		String className = element.getAttribute("class");
+		String bundleName = element.getContributor().getName();
+		try {
+			Class<?> clazz = Platform.getBundle(bundleName).loadClass(className);

```

Comment:
Why not using element.createExecutableExtension("class") ?

Reply:
yep, that's more correct. Changed in [8abb303](https://github.com/eclipse/tm4e/pull/184/commits/8abb303343e5ecd565bb52aca7d9efc4afdfdb16) commit.